presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonFunctions.java

```diff
@@ -88,38 +90,33 @@ public static JsonPath castCharToJsonPath(@LiteralParameter("x") Long charLength
         return new JsonPath(padSpaces(pattern, charLength.intValue()).toStringUtf8());
     }
 
-    @SqlNullable
     @ScalarFunction("is_json_scalar")
     @LiteralParameters("x")
     @SqlType(StandardTypes.BOOLEAN)
-    public static Boolean varcharIsJsonScalar(@SqlType("varchar(x)") Slice json)
+    public static boolean varcharIsJsonScalar(@SqlType("varchar(x)") Slice json)
     {
         return isJsonScalar(json);
     }
 
-    @SqlNullable
     @ScalarFunction
     @SqlType(StandardTypes.BOOLEAN)
-    public static Boolean isJsonScalar(@SqlType(StandardTypes.JSON) Slice json)
+    public static boolean isJsonScalar(@SqlType(StandardTypes.JSON) Slice json)
     {
         try (JsonParser parser = createJsonParser(JSON_FACTORY, json)) {
             JsonToken nextToken = parser.nextToken();
-            if (nextToken == null) {
-                return null;
-            }
+            checkCondition(nextToken != null, INVALID_FUNCTION_ARGUMENT, "Bad JSON value: %s", truncateIfNecessaryForErrorMessage(json));
 
             if (nextToken == START_ARRAY || nextToken == START_OBJECT) {
                 parser.skipChildren();
-                if (parser.nextToken() != null) {
-                    // Invalid JSON: trailing tokens
-                    return null;
-                }
+                checkCondition(parser.nextToken() == null, INVALID_FUNCTION_ARGUMENT, "Bad JSON value: %s", truncateIfNecessaryForErrorMessage(json)); // check no trailing token
```

Comment:
Put the comment on the previous line rather than at the end. We avoid end of line comments, especially for lines that are already very long.